Userland/Applications/Maps/UsersMapWidget.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2023, Bastiaan van der Plaat <bastiaan.v.d.plaat@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "UsersMapWidget.h"
+#include <AK/JsonParser.h>
+#include <LibDesktop/Launcher.h>
+
+UsersMapWidget::UsersMapWidget(Options const& options)
+    : MapWidget::MapWidget(options)
+{
+    m_marker_gray_image = Gfx::Bitmap::load_from_file("/res/graphics/maps/marker-gray.png"sv).release_value_but_fixme_should_propagate_errors();
+}
+
+void UsersMapWidget::get_users()
+{
+    // Start HTTP GET request to load people.json
+    HashMap<DeprecatedString, DeprecatedString> headers;
+    headers.set("User-Agent", "SerenityOS Maps");
+    headers.set("Accept", "application/json");
+    URL url("https://usermap.serenityos.org/people.json");
+    auto request = request_client()->start_request("GET", url, headers, {});
+    VERIFY(!request.is_null());
+    m_request = request;
+    request->on_buffered_request_finish = [this, request, url](bool success, auto, auto&, auto, ReadonlyBytes payload) {
+        m_request.clear();
+        if (!success) {
+            dbgln("Maps: Can't load: {}", url);
+            return;
+        }
+
+        // Parse JSON data
+        JsonParser parser(payload);
+        auto result = parser.parse();
+        if (result.is_error()) {
+            dbgln("Maps: Can't parse JSON: {}", url);
+            return;
+        }
+
+        // Parse each user
+        // FIXME: Handle JSON parsing errors
+        auto json_users = result.release_value().as_array();
+        for (size_t i = 0; i < json_users.size(); i++) {
+            auto const& json_user = json_users.at(i).as_object();
```

Comment:
I think you can leave out the `i`?
```
for (auto const& json_user_value : json_users) {
    auto const& json_user = json_user_value.as_object();
```

Reply:
JsonArray doesn't have a  begin() function that is needed for iterators right?

Reply:
Oh that's silly if we don't have that. But in that case, nevermind.

Reply:
We have `JsonArray::values()` so you could iterate that.